docs-markdown/src/controllers/cleanup-controller.ts

```diff
@@ -0,0 +1,308 @@
+"use strict";
+
+import * as vscode from "vscode";
+import { reporter } from "../telemetry/telemetry";
+import { window, workspace } from "vscode";
+import { readdir, stat, unlinkSync, readFile, writeFile, write } from "fs";
+import { join, resolve, dirname, basename } from "path";
+import { output } from "../extension";
+const jsyaml = require("js-yaml");
+const telemetryCommand: string = "applyCleanup";
+const replace = require('replace-in-file');
+const yaml = require('yaml')
+
+export function applyCleanupCommand() {
+    const commands = [
+        { command: applyCleanup.name, callback: applyCleanup },
+    ];
+    return commands;
+}
+
+export function applyCleanup() {
+    reporter.sendTelemetryEvent("command", { command: telemetryCommand });
+    const opts: vscode.QuickPickOptions = { placeHolder: "Cleanup..." };
+    const items: vscode.QuickPickItem[] = [];
+    items.push({
+        description: "",
+        label: "Single-valued metadata",
+    });
+    items.push({
+        description: "",
+        label: "Microsoft links",
+    });
+    items.push({
+        description: "",
+        label: "Capitalization of metadata values",
+    });
+
+    window.showQuickPick(items).then((selection) => {
+        if (!selection) {
+            return;
+        }
+
+        switch (selection.label.toLowerCase()) {
+            case "single-valued metadata":
```

Comment:
Are we using typescript 2? If so we could use the readonly modifier and start moving strings like this to immutable constants.